src/applications/travel-pay/components/complex-claims/pages/ExpensePage.jsx

```diff
@@ -0,0 +1,322 @@
+    key => EXPENSE_TYPES[key].route === expenseTypeRoute,
+  );
+
+  const expenseTypeFields = expenseType ? EXPENSE_TYPES[expenseType] : null;
+
+  const handleFormChange = (event, explicitName) => {
+    // Figure out the field name
+    const name = explicitName ?? event.target?.name ?? event.detail?.name; // rarely used, but safe to include
+
+    // Figure out the value (covers VA components + normal inputs)
+    const value =
+      event?.value ?? event?.detail?.value ?? event.target?.value ?? '';
+
+    setFormState(prev => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const handleOpenModal = () => setIsModalVisible(true);
+  const handleCloseModal = () => setIsModalVisible(false);
+
+  const validatePage = () => {
+    const requiredFields = ['date', 'amount']; // basic required fields
+
+    if (expenseType === 'Meal') requiredFields.push('vendor');
+    if (expenseType === 'Lodging') {
+      requiredFields.push('vendor', 'checkInDate', 'checkOutDate');
+    }
+    if (expenseType === 'Commoncarrier') {
+      requiredFields.push('transportationType', 'transportationReason');
+    }
+    if (expenseType === 'Airtravel') {
+      requiredFields.push(
+        'vendorName',
+        'tripType',
+        'departureDate',
+        'departureAirport',
+        'arrivalDate',
+        'arrivalAirport',
+      );
+    }
+
+    const emptyFields = requiredFields.filter(field => !formState[field]);
+    const isValid = emptyFields.length === 0;
+
+    setShowError(!isValid);
+    return isValid;
+  };
+
+  const handleContinue = () => {
+    const isValid = validatePage();
+    if (!isValid) return; // stop navigation if validation fails
+    navigate(`/file-new-claim/complex/${apptId}/review`);
+  };
+
+  const handleBack = () => {
+    navigate(`/file-new-claim/complex/${apptId}/choose-expense`);
+  };
+
+  return (
+    <>
+      <h1>
+        {' '}
+        {expenseTypeFields?.expensePageText
+          ? `${expenseTypeFields.expensePageText
+              .charAt(0)
+              .toUpperCase()}${expenseTypeFields.expensePageText.slice(
+              1,
+            )} expense`
+          : 'Unknown expense'}
+      </h1>
+      {showError && (
+        <p
+          ref={errorRef}
+          tabIndex={-1} // make focusable
+          style={{ color: 'red' }}
+        >
+          Please fill out all required fields before continuing.
+        </p>
+      )}
+      <p>
+        {`If you have multiple ${
+          expenseTypeFields.expensePageText
+        } expenses, add just one on this page. ` +
+          `You'll be able to add more expenses after this.`}
+      </p>
+      {expenseType === 'Meal' && (
+        <>
+          <VaTextInput
+            label="Where did you purchase the meal?"
+            name="vendor"
+            value={formState.vendor || ''}
+            required
+            onInput={handleFormChange}
+            error
+          />
+        </>
+      )}
+      {expenseType === 'Lodging' && (
```

Comment:
I feel like breaking these out into their own components could clean up the mark up in this component. It would allow us to have some concise conditional logic here. For example,
```js
{
  Lodging: <LodgingFields />,
  Meal: <MealFields />,
  Airfare: <AirfareFields />,
  // etc...
}[expenseType]
```
Those components could also export their own validation functions, so that could just be called here without having to worry about which fields need validation for that expense type.